test/bzl_version/bzl_ext_version_test.rs

```diff
@@ -0,0 +1,79 @@
+//! A test to ensure the rules_rust extension bzlmod core version is accurate.
+
+use runfiles::Runfiles;
+
+fn parse_rules_rust_version(text: &str) -> String {
+    let mut in_module = false;
+    let mut found_dep = false;
+    for line in text.split('\n') {
+        if in_module {
+            if line.ends_with(')') {
+                in_module = false;
+                continue;
+            }
+
+            if found_dep {
+                if let Some((param, value)) = line.rsplit_once(" = ") {
+                    if param.trim() == "version" {
+                        return value.trim().trim_matches(',').trim_matches('"').to_owned();
+                    }
+                }
+            }
+
+            if line.trim().starts_with("name = \"rules_rust\"") {
+                found_dep = true;
+                continue;
+            }
+
+            continue;
+        }
+
+        if line.starts_with("bazel_dep(") {
+            assert!(!found_dep, "Found `rules_rust` dep but couldn't determine version.");
+            in_module = true;
+            continue;
+        }
+    }
+    panic!("Failed to find rules_rust version in:\n```\n{}\n```", text);
+}
+
+fn parse_module_bazel_version(text: &str) -> String {
+    let mut found_module = false;
+    for line in text.split('\n') {
+        if found_module {
+            assert!(!line.ends_with(')'), "Failed to parse version");
+            if let Some((param, value)) = line.rsplit_once(" = ") {
+                if param.trim() == "version" {
+                    return value.trim().trim_matches(',').trim_matches('"').to_owned();
+                }
+            }
+        } else if line.starts_with("module(") {
+            found_module = true;
+            continue;
+        }
+    }
+    panic!("Failed to find MODULE.bazel version");
+}
+
+/// If this test fails it means `//:version.bzl` and `//:MODULE.bazel` need to
+/// be synced up. `//:version.bzl` should contain the source of truth.
+#[test]
+fn module_bzl_has_correct_version() {
+    let r = Runfiles::create().unwrap();
+
+    let core_version = {
+        let path = runfiles::rlocation!(r, std::env::var("CORE_MODULE_BAZEL").unwrap()).unwrap();
+        let text = std::fs::read_to_string(path).unwrap();
+        parse_module_bazel_version(&text)
+    };
+    let requested_version = {
+        let path = runfiles::rlocation!(r, std::env::var("MODULE_BAZEL").unwrap()).unwrap();
+        let text = std::fs::read_to_string(path).unwrap();
+        parse_rules_rust_version(&text)
```

Comment:
Shouldn't we also check that the module() call _also_ has the same version? I thought our plan was to always release these together?

Reply:
Ooops, yup! Added the extra test!

Reply:
Weird, I don't think I see it?

Reply:
https://github.com/bazelbuild/rules_rust/pull/3052/files#r1872478871

Reply:
For example, if I apply the following diff
```diff
diff --git a/extensions/prost/MODULE.bazel b/extensions/prost/MODULE.bazel
index 81a5121c..dc926be7 100644
--- a/extensions/prost/MODULE.bazel
+++ b/extensions/prost/MODULE.bazel
@@ -2,7 +2,7 @@

 module(
     name = "rules_rust_prost",
-    version = "0.55.0",
+    version = "0.54.1",
 )

 bazel_dep(
```
This test fails
```
INFO: From Testing //private/tests/bzlmod:bzl_version_test:
==================== Test output for //private/tests/bzlmod:bzl_version_test:

running 1 test
test module_bzl_has_correct_version ... FAILED

failures:

---- module_bzl_has_correct_version stdout ----
thread 'module_bzl_has_correct_version' panicked at external/rules_rust~/test/bzl_version/bzl_version_test.rs:36:5:
assertion `left == right` failed: //:version.bzl and //:MODULE.bazel versions are out of sync
  left: "0.55.0"
 right: "0.54.1"
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    module_bzl_has_correct_version

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.01s
```